src/cimplog-rdk-logger.c

```diff
@@ -34,21 +34,27 @@ void __cimplog(const char *module, int level, const char *msg, ...)
 
     if( !init_done )
     {
-        RDK_LOGGER_INIT();
         rdk_logger_module = rdk_logger_module_fetch();
         if( NULL == rdk_logger_module )
         {
-            fprintf(stderr, "\nERROR: RDK Logger not integrated for this module !!!\n");
+            //If RDK logger is not integrated, give the information on the 'stderr' to use the weak implementation of
+            //'const char *rdk_logger_module_fetch(void)' method, through which the rdk_logger_module can be obtained.
+            fprintf(stderr, "\nERROR: RDK Logger not integrated for this module - '%s' !!!\n",module);
             fprintf(stderr, " Provide cimplog method \"const char *rdk_logger_module_fetch(void)\" to get log prints !!\n");
-	          exit(0); // Not using RDKLogger is an Error. Terminate!
+        }
+        else
+        {
+            RDK_LOGGER_INIT();
```

Comment:
init_done should be flagged only on RDK_LOGGER_INIT() case and not in the case where rdk_logger_module is not defined.

Reply:
Hey @shilpa24balaji,
Per design of the CIMPLog integration on RDKB, the rdk_logger_module will be defined in the component that is linking this library.
If it is not defined, it is never defined for the life time of that build. So there is no point in checking the on every print and thats why we have init_done set to true.

Infact, the flag shd be named init_once..

Reply:
rdk_logger_init makes sense only when logger module is defined as only then RDK_LOG() is called. But yes if logger module is not defined then we need not check again, so the code is fine.